### PR TITLE
Add --pubsubIdAttribute beam option for dedup when reading pubsub

### PR DESCRIPTION
**Pull request by relud**
for [Bug 1627758](https://bugzilla.mozilla.org/show_bug.cgi?id=1627758)

~needs tests~

### CODE REVIEW

File: ingestion-beam/src/main/java/com/mozilla/telemetry/io/Read.java
Comment (jklukas) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/io/Read.java:54]: `getIdAttribute` is marked as `@Nullable` in PubsubIO, which means `idAttribute` is allowed to be null. I think this means it's also possible to pass in `null` to `withIdAttribute`, so this logic is hopefully unnecessary.

Comment (relud) [reply ingestion-beam/src/main/java/com/mozilla/telemetry/io/Read.java]: yep, works as expected when null.

**Comment by relud** [comment]
@whd this is going to be a pre-template option, because the SDK doesn't accept a `ValueProvider` here.

**Comment by whd** [comment]
> @whd this is going to be a pre-template option, because the SDK doesn't accept a `ValueProvider` here.

Can you add it to https://github.com/mozilla/gcp-ingestion/blob/master/ingestion-beam/bin/build-template#L55 so it gets picked up automatically?

**Comment by relud** [comment]
> add it to https://github.com/mozilla/gcp-ingestion/blob/master/ingestion-beam/bin/build-template#L55

done

**Review by jklukas — APPROVED** [review]

**Comment by jklukas** [comment]
Up to @whd as to whether we should delay merging right now, or if you'd like this to go to staging right away.

**Comment by whd** [comment]
> Up to @whd as to whether we should delay merging right now, or if you'd like this to go to staging right away.

Let's merge this on Monday. The tentative plan:

1. merge on the 25th, auto-deploy to stage, observe stage
2. deploy prod (telemetry decoder only) before UTC 26th
3. observe performance of the decoder, run until at least UTC 27th
4. inspect dedup performance from e.g. pbe for the 26th
5. decide whether to roll forward the structured decoder or roll back the telemetry decoder
6. assuming roll forward, continue deploys as per usual

**Comment by whd** [comment]
I'm holding off on testing this until an unrelated support case with Google is resolved.

**Comment by whd** [comment]
Looks like the issue is resolved, so we can set this test up next week.

**Comment by whd** [comment]
I'm pushing this test out to next week (UTC Feb 17) due to https://github.com/mozilla/gcp-ingestion/issues/1560 and because I'll be out after Wednesday this week.

**Comment by whd** [comment]
I'm planning to merge this after https://github.com/mozilla/gcp-ingestion/pull/1566 is deployed.

**Pull request merged**
